src/openforms/js/components/admin/form_design/FormLogic.js

```diff
@@ -385,6 +341,137 @@ const Rule = ({
 
         <ActionSet name="actions" actions={actions} onChange={onChange} errors={errors.actions} />
       </div>
+    </>
+  );
+};
+
+RuleBody.propTypes = {
+  isCreate: PropTypes.bool.isRequired,
+  isAdvanced: PropTypes.bool.isRequired,
+  jsonLogicTrigger: PropTypes.object,
+  actions: PropTypes.arrayOf(PropTypes.object),
+  displayAdvancedOptions: PropTypes.bool,
+  description: PropTypes.string,
+  triggerFromStepIdentifier: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  errors: PropTypes.object,
+  mayGenerateDescription: PropTypes.bool.isRequired,
+};
+
+const Rule = ({
+  isCreate,
+  _logicType,
+  description,
+  _mayGenerateDescription,
+  order,
+  jsonLogicTrigger,
+  triggerFromStep: triggerFromStepIdentifier,
+  actions,
+  isAdvanced,
+  onChange,
+  onDelete,
+  errors = {},
+}) => {
+  const intl = useIntl();
+  const [displayAdvancedOptions, setDisplayAdvancedOptions] = useState(false);
```

Comment:
I think we could do the broken-trigger-from-step detection here and then you can immediately call `setDisplayAdvancedOptions(true)` if that's the case (or use that state as initial state value :thinking: )

Reply:
Mhmmm I wanted to have as much logic as possible inside the error boundary, so that if something unexpected goes wrong it doesn't break the page but the action to delete is still visible :thinking: Maybe I can pass `setDisplayAdvancedOptions` to `RuleBody`?

Reply:
![Screenshot from 2023-09-08 12-08-39](https://github.com/open-formulieren/open-forms/assets/19154114/1d94fe4f-3968-4afb-8f6a-de07783f97bc)

Reply:
is this warning also done with error boundary then?

> Maybe I can pass setDisplayAdvancedOptions to RuleBody?

It's an option, but then you'll have to rely on `useEffect` I think and it may conflict with the user click actions that update the state :thinking: 

Error boundaries are really a bit of a last resort btw, it's good to have them because you don't know what will break, but if you can detect error situations and prevent crashes that way, that's always preferred. So that's why I suggested this approach :)

Reply:
> is this warning also done with error boundary then?

No, this error is done by checking if `triggerFromStepIdentifier` is set but `triggerFromStep` is null! 

But what I meant is: the errors tend to come from the logic that is now moved into the `RuleBody` component and wrapped in an error boundary.
If I move the code that checks for the trigger step back to `Rule`, if it breaks unexpectedly it will break the whole page because it will be outside of the error boundary.

Reply:
I think I understand.

Could we maybe perform the check inside the advanced config (the component behind the cog) and force that to be open regardless of the toggle state/prop if there's an error? So then the actual checks are all contained inside the error boundary but we get nicer feedback.